docs/guide/styles.md

```diff
@@ -1,15 +1,320 @@
+```
+
+Notice the additional space around the text:
+
+```{.textual path="docs/examples/guide/styles/padding01.py"}
+```
+
+You can also set padding to a tuple of two integers which will apply padding to the top/bottom and left/right edges. The following example sets padding to `(2, 4)` which adds two rows to the top and bottom of the widget, and 4 columns to the left and right of the widget.
+
+```python title="padding02.py" hl_lines="22"
+--8<-- "docs/examples/guide/styles/padding02.py"
+```
+
+Compare the output of this example to the previous example:
+
+```{.textual path="docs/examples/guide/styles/padding02.py"}
+```
+
+You can also set padding to a tuple of four values which applies padding to each edge individually. The first value it the padding for the top of the widget, followed by the right of the widget, then bottom, then left.
+
+### Border 
+
+The [border](../styles/border.md) style draws a border around a widget. To add a border set `syles.border` to a tuple of two values. The first value is the border type, which should be a string. The second value is the border color which will accept any value that works with  [color](../styles/color.md) and [background](../styles/background.md).
+
+The following example adds a border around a widget:
+
+```python title="border01.py" hl_lines="22"
+--8<-- "docs/examples/guide/styles/border01.py"
+```
+
+Here is the result:
+
+```{.textual path="docs/examples/guide/styles/border01.py"}
+```
+
+There are many other border types. Run the following from the command prompt to preview them.
+
+
+```bash
+textual borders
+```
+
+### Outline
+
+[Outline](../styles/outline.md) is similar to border and is set in the same way. The difference is that outline will not change the size of the widget, and may overlap the content area. The following examples sets an outline on a widget:
+
+```python title="outline01.py" hl_lines="22"
+--8<-- "docs/examples/guide/styles/outline01.py"
+```
+
+Notice how the outline overlaps the text in the widget.
+
+```{.textual path="docs/examples/guide/styles/outline01.py"}
+```
+
+Outline can be useful to emphasize a widget, but be mindful that it may obscure your content.
+
+### Box sizing
+
+When you set padding or border it reduces the size of the widget's content area. In other words, setting padding or border won't change the width or height of the widget.
+
+This is generally desirable when you arrange things on screen as you can add border or padding without breaking your layout. Occasionally though you may want to keep the size of the content area constant and grow the size of the widget to fit padding and border. The [box-sizing](../styles/box_sizing.md) style allows you to switch between these two modes.
+
+If you set `box_sizing` to `"content-box"` then space required for padding and border will be added to the widget dimensions. The default value of `box_sizing` is `"border-box"`. Compare the box model diagram for content box to the previous box model diagram.
```

Comment:
Only thing I'd say here is "the previous box model diagram" is so far back up the page I'd almost forgotten about it.
When I read this my instinct is to want to compare them. I'd probably like to see both of them side by side, or at least in tabs that I can click between instead of having to scroll away back up to spot the difference.

Reply:
I'll add a link